### ISSUE TITLE
Changeling reworks- What is to be done? 

### ISSUE THREAD

**Issue by icantthinkofanameritenow** [d3athrow/vgstation13#10439]
So since my first round as a Changeling after the anti-monkeyman PR I made I've learned that the Changeling's main flaw is simple- it has very little in the way of advantages in a straight-up fight. Sure, its stings are dangerous and if it's caught you alone in maint you're done for. But if the ling can't reliably lure people away via RPing (assuming you can ever get someone to play along with that) or somehow abduct them while there's nobody watching (i.e. rarely if ever between the AI, security cameras, and the inevitable bystanders who will immediately call for Security even if the guy you're dragging off is incapable of it). it's still reduced to pseudo-extended due to the fact that its one main advantage (regenerative stasis) will be rendered irrelevant since the second someone knows you're a ling they'll destroy your body right away. 

Thus, I would like to either port more aggressive powers from other servers or make some of my own. The arm blade from TG is the most obvious of these, since it's robust and solves the problem of "how am I going to kill these people when I don't have enough evolution points for death sting and lack good opportunities to kill stealthily?" 

Other things we could consider is having the Extend Range power actually make the stings act like projectiles, or a bio-gun akin to the Hivehand capable of dealing toxin damage at the expense of not being able to use that limb and requiring chemicals to maintain it. The important thing is that they should have direct-killing functions along the lines of a Traitor's esword and revolver, and make Changeling be more like the Thing as opposed to a parapen-toting maint stalker that rarely does much genuine antagonizing. 

Some people have suggested making it a team mode where each changeling can convert others into changelings, but I feel that this would make them go from boring to impossible to deal with. But if someone can explain how the lings created in this manner wouldn't be undefeatable and could be viably stopped, they are free to do so.

**Comment by icantthinkofanameritenow** [comment]
While it might be a bit derivative, I've also noted that the Chiropteran Screech power was originally used for Changelings (under a different name), so it might be possible to do that as well. Some servers also seem to have an EMP generation power for lings as well, and while I'm a bit reluctant to add that it might be balanced via a fairly high evolution point cost, a high chemical cost, or both.

**Comment by AutismSpider** [comment]
keep the screech to vamps. Unless going take out cheap ass death sting or turn death sting back into lex

**Comment by icantthinkofanameritenow** [comment]
@AutismSpider

Merely a suggestion. And of course death sting assumes you can get enough evolution points to afford it in the first place.

And need I remind you that lexorin is so trivially simple to counter and lacking in actual killing ability that  it's use in death sting made it a waste of evolution points?

**Comment by Blithering** [comment]
I think they'd be more inclined to go loud if they had a range of options for regenerating. Currently everyone knows to remove the head, gib, burn or monkeyspike lings to prevent them regenerating. If they could choose between regen stasis, becoming a headspider or laying down some form of cocoon ahead of time to hatch out of in case of death it would create more chances to come back after a fuckup.

**Comment by icantthinkofanameritenow** [comment]
@Blithering

And we do have the headspider sprite already, so that could be easily implemented (relatively speaking) as a last-ditch measure that could be triggered only after decapitation or something along those lines. More importantly, they'd need a new identity after coming back that way, since if they return in their original appearance everyone will be gunning for them. Given that very few lings seem to be able to absorb even one genome in a given round, perhaps either discounting the DNA extract sting or making it free might help in that regard. 

I personally think another issue is that lings have no way to deal with stuns like vampires can, so one taser shot means that it's all over for them if they didn't splurge enough to go into horror form. Rather than just copypasting the vampire Rejuvenate ability onto them, I was thinking it might be possible for them to gain an ability that passively reduces all stun durations by about 1/4 or so of their original time. So a 30 second-stun might only last a little over 20 seconds, for example. 

Ranged attacks in general are also an issue for them, since their main methods of killing (barring getting a gun or a weapon from elsewhere, which isn't always a guarantee and may draw attention from Security even if they haven't been confirmed to be changelings)  all force them to be within 2 spaces of the target, and as I found out the hard way simply staying near someone for too long is enough to set off people's alarm bells. Hence my suggestion for a "bio-gun" of sorts in my first post. 

I'm sort of visualizing it as an energy crossbow minus the radiation effect, or at least with a somewhat similar role (e.g. difficult to detect ranged stun that also does a little damage). On that note, the "delayed poison sting" from Polaris might also be a useful addition, since it can be used to deal damage to someone without the user being directly implicated as the source of said damage.

**Comment by Blithering** [comment]
Sounds good, so long as the ranged stuns aren't spammable. Noone likes to fight a guy who has an ebow shoved down his throat that he can use while stunned. Maybe just rework the current ranged sting system into a true projectile.

**Comment by icantthinkofanameritenow** [comment]
@Blithering 
Agreed, on both counts. 

For the biogun concept, I was figuring that maybe 10 chems consumed per shot would be reasonable enough, and I had no intention of it being usable while stunned. Plus, like the Hivehand it can't be dropped unless you toggle it back off and it would slow chemical regeneration while active to keep it from being spammed too much.

**Comment by icantthinkofanameritenow** [comment]
Someone on the current thread mentioned the possibility of changelings being able to convert other people into changelings like in The Thing. I'm really not convinced that's a good idea because of the potential to turn every absorbed victim into a new changeling, but I'd like a second opinion about it. 

Assuming that such a mechanic wouldn't be stupidly overpowered in itself, how could it be implemented in a reasonable manner?

**Comment by Blithering** [comment]
I did think it would be interesting to have a 'reanimation sting' which turned husks into zombie like minions. Mainly as a distraction to let the ling get away.

**Comment by icantthinkofanameritenow** [comment]
That does doesn't seem to be going over too well- I think people want it to be a complete copy of The Thing, balance and fun for others be damned.

It's like I'm working at cross-purposes here- I want to make them into primarily solo stealthy killers which can still put up a fight if they must go loud,. not a pseudo Revolution mode where every dead crew member becomes a Changeling themselves. At that point you may as well stop calling them changelings entirely since they would play nothing like they do now. 

Or do they want Changelings to be unstoppable instead of simply boring?

**Comment by ComicIronic** [comment]
I don't think they mean every corpse as much as having the power to split with enough biomass. It would be an interesting dynamic to spawn an alt and then sneak around the station together stealing identities. 

Currently lings have the coop downside of munching each other - voluntary splitting would overcome that somewhat.

**Comment by Blithering** [comment]
How about a lesser horror form, that can be activated and deactivated without alerting the whole station, but costs one evolution point per use? That way even starting lings can fight back when backed into a corner, but are discouraged from going loud from the start.

**Comment by clusterfack** [comment]
I think we should have the Bay style genome stealing, where you go near a person and completely undetectably steal their genome. This gives a much weaker reward than sucking off a crewmember, and lets you collect identities without forcing you to kill people.

And the fact you can collect multiple identities also allows you to use transformations more often, allowing us the **possibility** of adding a changeling test down the line, to increase tension for the mode.

If we make changeling a mode where people HAVE to kill to get genomes then it will inevitably become a death match.

**Comment by icantthinkofanameritenow** [comment]
@ComicIronic 
No, they were pretty clear that they would be the exact same corpse brought back to life as a changeling. And that it should be a team mode where munching other lings would be completely taboo if not impossible. 

The fact that the station would be completely overwhelmed by changeling hordes as a result of this is either lost to them, or simply not something they care about. I'm growing increasingly sick of hearing about "MUH THE THING", and wondering if any of these people have ever considered what their changes would actually mean in practice. And apparently pointing out that even Polaris' changeling which they so love is just ours with arm blades and some RP powers just flies over their heads. 

@clusterfack 
DNA extract sting does that to an extent, but grants no evolution points and doesn't count for objectives. Without a way to actually incapacitate or kill, it's little more than an evolution point sink even with the discount I made a while back. Better to make it free and count for evolution points and genomes, and have traditional absorbing just give double evolution points and/or a chance to re-select powers.

**Comment by clusterfack** [comment]
@icantthinkofanameritenow 
I agree with you fully I think, yes we chould change the DNA extract sting to both be free, and to give smaller amounts of evolution points. I don't think it should count for objectives though and we should increase the point value of absorbing real live humans.

**Comment by icantthinkofanameritenow** [comment]
@clusterfack 
But then we may run the problem of them using it to just go straight for horror form or another high-point cost power. Perhaps some powers could require absorptions as well as evolution points? 

Even then, I feel that people will need a LOT of convincing. They don't want Changeling fixed, they just want a completely different mode that I have every reason to believe they will despise the moment they realize how badly it works for them.

**Comment by clusterfack** [comment]
I don't think it will be possible to go straight for the high point cost powers very easily, but it does all depend on the way the point costs are balanced. I think it should be balanced around the idea of stealing identities which gives you the (effectively) minor powers that let you finally absorb people giving you the points to get the (effectively) major powers.

**Comment by icantthinkofanameritenow** [comment]
So how would that work? 1 point for extracts, and 3 for absorptions? Because on highpop that still translates into fairly easy acquisitions of high-tier powers unless we jack up all the costs at once.

**Comment by clusterfack** [comment]
I mean yes if you want to hold to all the current point values for all other powers right now (though there is absolutely no reason we are forced to). But I think the point values for all the other powers right now are balanced around the fact you cant get genomes from ANYTHING other than absorbing.

I think some of the powers, ie the deadly (death sting) and op (horror form), should be clearly differentiated in terms of point cost from the other minor powers that help you get absorptions.

I think we should think of it more along these lines
~1-5 identities, 1-3 minor powers unlocked
~6-10 identities, 3-5 minor powers unlocked, enough to start generally absorbing people

~3 absorptions, 1 major power
~6 absorptions, 2 major powers or something like horror form

**Comment by Blithering** [comment]
How about you can't absorb and extract the same genome? You have to balance between easy early boosts or late game power.

**Comment by clusterfack** [comment]
It's not a terrible suggestion but I think it disproportionately hurts newbies who will tend to play it safe. 

As a general goal I would want the skill gap between **relatively** new players and those who have played the game to the point of mastery to not be insurmountable so that both could still be effective.

**Comment by AutismSpider** [comment]
Give me power to shit out spiderlings/Head spider if i happen to get my head chop off.

Blade hands would be nice

**Comment by icantthinkofanameritenow** [comment]
@clusterfack 
That sounds like a good idea, if one that's beyond my ability to code myself.  I'm more than a bit worried that the community will dislike it enough to keep any changes from going through though- they're so fixated on "reworking" Changeling that they care nothing about how their reworks would actually behave in game or how fun it would actually be.

**Comment by icantthinkofanameritenow** [comment]
So now I'm starting to think the best idea is to port most if not all of Polaris's changeling features. But I don't know how. Is there a manual for how to port shit?

**Comment by Shadowmech88** [comment]
> Is there a manual for how to port shit

1) Copy and paste everything you want to port from their code into our code.
2) Fix the inevitable compile errors by either removing references to things that we don't have or porting those things as well.
3) Continue patching holes until it compiles.
4) Code as normal from there.

**Comment by icantthinkofanameritenow** [comment]
@Shadowmech88 
Tried doing it with Polaris. Turns out their code is so completely different from ours that the entire gamemode code will need to be rewritten for a port to work. And I'm nowhere near skilled enough to do that myself.

**Comment by Shadowmech88** [comment]
So tighten the scope.

Write a list of ways in which Polaris's changeling is different from ours.
Then, one by one, implement those features into our changelings.

**Comment by icantthinkofanameritenow** [comment]
@Shadowmech88 

Here's the list of the "must have" powers, then. But I don't know how to implement them. 
-Arm blades (an energy sword equivalent that vanishes when "dropped")
-Fabricate Clothing (create a fake inventory for self to look like the person you transformed into, complete with fake ID)
-Organic Space Suit (pressure and temp resistance, spaceworthy but needs internals, unless you have...)
-Self-Respiration (lose the need to breathe, akin to the No Breathing superpower in genetics only it also keeps you from being affected by plasma and harmful gases too)
-Bioelectrogenesis (think stungloves that can never be stripped)
-Electric lockpick (opens locked doors, seems too much like an emag IMO)
-Delayed Toxic Sting (injects a poison that causes toxin damage after a few minutes)
-Chitin Space Armor (like the organic space suit, but is also armor as well)
-Dissonant Shriek (instant EMP)

More info is here: http://ss13polaris.com/wiki/doku.php?id=game:guides:changeling 
Keep in mind when looking at prices that the only way for lings to get more evolution points is by absorbing other lings. Absorbing anyone else only lets them re-pick their powers.

**Comment by icantthinkofanameritenow** [comment]
Working on an arm blade now, will make a PR for it when it's ready.

**Comment by icantthinkofanameritenow** [comment]
First one being addressed in #10458

**Comment by icantthinkofanameritenow** [comment]
Which of these should be next up? 

Also, planning to jack up the cost of death sting slightly per popular demand.

**Comment by The-Albinobigfoot** [comment]
> Chitin Space Armor (like the organic space suit, but is also armor as well)

Consider ditching the space-worthy aspect and making it armor only. Otherwise, who the fuck is going to bother with the organic space suit anymore?

**Comment by Kurfursten** [comment]
Changeling is no longer in, make a new issue if you want to discuss modern ling reform
